File: src/ELibrary.Frontend/src/app/modules/shared/domain/dtos/library/author/updateAuthorRequest.ts
import { Author } from "../../../models/library/author";

export interface UpdateAuthorRequest {
    id: number;
    name: string;
    lastName: string;
    dateOfBirth: Date;
}
export function mapAuthorToUpdateAuthorRequest(author: Author): UpdateAuthorRequest {
    return {
        id: author.id,
        name: author.name,
        lastName: author.lastName,
        dateOfBirth: author.dateOfBirth
    }
}

File: src/ELibrary.Backend/LibraryApi/Domain/Dto/Library/Author/UpdateAuthorRequest.cs
﻿namespace LibraryApi.Domain.Dtos.Library.Author
{
    public class UpdateAuthorRequest
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string LastName { get; set; }
        public DateTime DateOfBirth { get; set; }
    }
}